src/Core/IdentityServer/BaseRequestValidator.cs

```diff
@@ -259,11 +274,63 @@ protected async Task BuildErrorResultAsync(bool twoFactorRequest, T context, Use
             return new Tuple<bool, Organization>(individualRequired || firstEnabledOrg != null, firstEnabledOrg);
         }
 
+        private async Task<bool> RequiresSsoAuthAsync(User user, string grantType)
```

Comment:
Let's consider renaming this to be more explicit as to the condition and return value's definition/meaning. This name would imply it returns `true` if SSO is required, however the opposite is true, therefore perhaps it should be flipped in return value, OR, renamed to `SsoAuthNotRequiredAsync` or something like that?

Reply:
Will update!

Reply:
Updated. `IsValidAuthTypeAsync` covers whats happening and comments give more context.